### PR TITLE
My homework 

### PR DESCRIPTION
**Pull request by massoodbarsa**
week 1
week 2

### CODE REVIEW

**Comment by rts-rob** [comment]
@massoodbarsa - Can you add the help file "help.txt" to this PR please?

File: week2/index.js [hunk 1]
```diff
@@ -58,6 +101,78 @@ function listTodos() {
...
+function update() {
+  fs.readFile('todo.txt', 'utf8', function(err, data) {
+    if (err) {
+      throw err;
+    }
+    todo = splitStringByNewline(data)
+    let number = num()
+    todo[number - 1] = options[2]
+    let joinedLines = todo.join('\r\n').trim();
+    fs.writeFile('todo.txt', joinedLines);
```
Comment (rts-rob) [inline week2/index.js]: You mean "fs.writeFileSync"? When I run your code I get:
`(node:1614) [DEP0013] DeprecationWarning: Calling an asynchronous function without callback is deprecated.`

File: week2/index.js [hunk 2]
```diff
@@ -58,6 +101,78 @@ function listTodos() {
...
+  })
+}
+
+//update a todo item with new text
+function update() {
+  fs.readFile('todo.txt', 'utf8', function(err, data) {
+    if (err) {
+      throw err;
+    }
+    todo = splitStringByNewline(data)
```
Comment (rts-rob) [inline week2/index.js]: You haven't yet defined `todo` - best practice is to use `let todo = ...` here.

File: week2/index.js [hunk 3]
```diff
@@ -4,14 +4,56 @@ var options = process.argv.slice(2);
...
+    remove()
+    break;
+  case 'reset':
+    reset()
+    break;
+  case 'update':
+    update()
+    break;
+  default:
+    showHelp();
```
Comment (rts-rob) [inline week2/index.js]: Why not just `help()` as above?
Even better - how can we change this case to execute for both `help` and the default?
(DRY - don't repeat yourself)

**Comment by rts-rob** [comment]
Good work here! I ran this through `eslint` and it raised a lot of warnings. Let's talk in class or via slack about how you can quickly format your code to remove those.

File: week2/index.js [hunk 4]
```diff
@@ -4,14 +4,56 @@ var options = process.argv.slice(2);
...
+
+//find number after instruction
+function num() {
+  let number = parseInt(options[1])
+
+  //avoid type anything but numbers
+  if ((typeof number === 'number' && !isNaN(number))) {
+    return number
+
+  } else if (options[1] == null) {
```
Comment (jasongwartz) [inline week2/index.js]: Careful with this `options[1] == null` comparison. While it does work right now, the actual value of an out-of-bounds array element is undefined, not null. So `options[1] === null` would not do what you want, whereas `options[1] === undefined` does, and is more likely to be what you intended for this line to mean.

**Pull request closed without merging**